src/tunnel.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef TUNNEL_H_
+#define TUNNEL_H_
+
+#include "comm/forward.h"
+#include "HttpRequest.h"
```

Comment:
Please forward-declare HttpRequest class below instead of including the whole header.